### DISCUSSION TITLE
Wishlist

### DISCUSSION THREAD

**Discussion by caizhongang** [open-mmlab/mmhuman3d#47] [Ideas]
### Welcome notes
Please leave a comment on which new method/feature/dataset you would like us to add in MMHuman3D.
The most upvoted suggestions will be prioritized!

### Updates
- [Requests](https://github.com/open-mmlab/mmhuman3d/discussions/47#discussioncomment-1863714) gathered after the release of v0.3.0.
- [Requests](https://github.com/open-mmlab/mmhuman3d/discussions/47#discussioncomment-2166743) gathered after the release of v0.4.0.
- [Requests](https://github.com/open-mmlab/mmhuman3d/discussions/47#discussioncomment-2485652) gathered after the release of v0.5.0.
- [Requests](https://github.com/open-mmlab/mmhuman3d/discussions/47#discussioncomment-2691157) gathered after the release of v0.6.0.
- [Requests](https://github.com/open-mmlab/mmhuman3d/discussions/47#discussioncomment-3142536) gathered after the release of v0.7.0 and v0.8.0.

### Links
Please refer to our [Roadmap](https://github.com/open-mmlab/mmhuman3d/discussions/48) for upcoming updates.

**Comment by caizhongang** [comment]
In the first three weeks after the release of v0.3.0, these requests have stood out (ordered by popularity):

- Add new SOTAs: PARE and GLAMR **(update: PARE is added in [v0.7.0](https://github.com/open-mmlab/mmhuman3d/releases/tag/v0.7.0))**
- Support SOTAs for SMPL-X (Frankmocap, PIXIE, Pose2Pose) 
- Support for clothed humans
- Support .fbx output

**Reply by lucasjinreal** [reply]
small suggestions, if fbx can drive mixamo charactors would be better.

**Reply by caizhongang** [reply]
@jinfagang Thank you for your suggestion, you may want to start a new comment below instead of commenting on my comment so people can upvote your idea!

**Reply by korotetskiiba** [reply]
fbx output ❤️

**Reply by rlleshi** [reply]
When could `fbx` output be supported? Is there a rough timeline?

**Comment by lucasjinreal** [comment]
Does it consider support some acceleration method in MMDeploy, for instance, add a more powerful but more faster detector base? Or accelerated pose model from mmpose?

**Reply by caizhongang** [reply]
@jinfagang Thank you for you suggestion! We are looking into collaboration with other codebases such as MMDeploy and MMRazor. However, we do not have a timeline yet. We will keep you updated.

**Comment by socketing** [comment]
很希望可以支持rgbd的支持。可以从深度数据上或者点云数据上实现对模型的参数化生成。可以更好的对已生成的三维模型进行尺寸估算

---

Translation: hope to see support for rgbd. Parametric model from depth / point cloud data give better dimension estimation.

**Reply by caizhongang** [reply]
@socketing 感谢你的建议，基于点云的方法已经在我们的[Roadmap](https://github.com/open-mmlab/mmhuman3d/discussions/48#discussion-3763206)中了，相信今年上半年就能加入代码库。

---

Translation: we are developing point cloud-based methods, that are likely to be added in 2022 H1. Please check our [Roadmap](https://github.com/open-mmlab/mmhuman3d/discussions/48#discussion-3763206) for updates.

**Comment by lucasjinreal** [comment]
支持一下HybrIK推理

---

Translation: support HybrIK inference

**Comment by rlleshi** [comment]
New Feature: Save the predictions (bboxes, poses, meshes) in pkl format

**Comment by littlejiumi** [comment]
Great work!
Hope to support [ROMP](https://github.com/Arthur151/ROMP): [ICCV 2021] Monocular, One-stage, Regression of Multiple 3D People

**Comment by Gzzgz** [comment]
Hope to support MANO

**Comment by caizhongang** [comment]
After the release of v0.4.0, the team has taken notes of the following requests:
- Support point cloud-based methods, suggested by @socketing 
- More SMPL SOTA: ROMP, suggested by @littlejiumi 
- Save inference results, suggested by @rlleshi **(update: added in [v0.7.0](https://github.com/open-mmlab/mmhuman3d/releases/tag/v0.7.0))**
- Collaboration with other codebases (MMRazor, MMDeploy), suggested by @jinfagang

**Reply by lucasjinreal** [reply]
does there any examples/tutors on deploy any mesh3d human models?

**Reply by caizhongang** [reply]
> - Save inference results, suggested by @rlleshi

@jinfagang we will add a feature to save SMPL inference results, from which you can obtain human mesh vertices and surfaces. You may also refer to [this tutorial](https://github.com/open-mmlab/mmhuman3d/blob/main/docs/visualize_smpl.md) for visualization of SMPL mesh.

**Comment by JackieXuu** [comment]
Hope to train SPIN(or other methods) using our own datasets(like openpose-based gt). For SPIN method which needs fitting files, I hope that if we use other own datasets, it can generate initial fitting parameters.

**Comment by rawalkhirodkar** [comment]
Thank you for the work.
Can we have temporal fitting methods like [DynaBOA](https://github.com/syguan96/DynaBOA)?

**Comment by Kayce001** [comment]
Hope to support mobile deployment.

**Reply by rlleshi** [reply]
In other words, `fbx` output. Then you can convert that as required, for example to `dae` and deploy it on an IPhone

**Comment by imabackstabber** [comment]
Please Add MANO.

**Reply by caizhongang** [reply]
MANO supported with ExPose in [v0.9.0](https://github.com/open-mmlab/mmhuman3d/releases/tag/v0.9.0). ;)

**Comment by caizhongang** [comment]
After the release of v0.5.0, the team has taken note of the following requests:
- Add tools to generate intial fitting parameters for SPIN-like methods, suggested by @ZhengtianXu 
- Parametric model beyond body: MANO @imabackstabber @Gzzgz 

The following requests have already been taken note in previous request collections:
- Mobile deployment, suggested by @Kayce001 
- `fbx` output, mentioned by @rlleshi

**Comment by ykk648** [comment]
[PyMAF](https://github.com/HongwenZhang/PyMAF)
[Relative_Human](https://github.com/Arthur151/Relative_Human)

**Comment by caizhongang** [comment]
After the release of v0.6.0, the team has taken note of the following requests:
- More SMPL methods ([PyMAF](https://github.com/HongwenZhang/PyMAF) and [Relative_Human](https://github.com/Arthur151/Relative_Human)), suggested by @ykk648 

We would like to seek community's kind contribution to MMHuman3D by reproducing SOTA methods! Please start a discussion.

**Reply by akk-123** [reply]
wish support new sota PIXIE

**Comment by JiahongWu1995** [comment]
Please Add SmoothNet.

**Reply by caizhongang** [reply]
SmoothNet has been added in [v0.8.0](https://github.com/open-mmlab/mmhuman3d/releases/tag/v0.8.0) :)

**Comment by Serdnad** [comment]
+1 for temporal methods, [DynaBOA](https://github.com/syguan96/DynaBOA) in particular.

Would also be great to see support for [STAR](https://star.is.tue.mpg.de/), meant to be "a drop in replacement for SMPL" partially made by one of the original SMPL researchers.

**Reply by caizhongang** [reply]
STAR is added in [v0.9.0](https://github.com/open-mmlab/mmhuman3d/releases/tag/v0.9.0). ;)

**Comment by ykk648** [comment]
[shapy](https://github.com/muelea/shapy)

**Comment by kristijanbartol** [comment]
[STRAPS](https://www.bmvc2020-conference.com/assets/papers/0081.pdf) and [HierProb3D](https://github.com/akashsengupta1997/HierarchicalProbabilistic3DHuman). These are one of the only methods for _accurate_ shape estimation in-the-wild (most of the other methods often just use an average shape or estimate shape only based on 2D joint estimations).

**Reply by caizhongang** [reply]
Great suggestion, may also look at [Shapy](https://shapy.is.tue.mpg.de/).

**Comment by caizhongang** [comment]
After the release of v0.7.0 and v0.8.0, the team has taken note of the following requests:

- Temporal fitting methods such as [DynaBOA](https://github.com/syguan96/DynaBOA), suggested by @Serdnad 
- Shape estimation methods such as [STRAPS](https://www.bmvc2020-conference.com/assets/papers/0081.pdf), [HierProb3D](https://github.com/akashsengupta1997/HierarchicalProbabilistic3DHuman), and [Shapy](https://shapy.is.tue.mpg.de/), suggested by @ykk648 and @kristijanbartol 

Community's contributions to MMHuman3D by reproducing SOTA methods are highly welcome! Please start a discussion.

**Reply by ChennyDeng** [reply]
Cant wait to see DynaBOA!

**Comment by axhiao** [comment]
Hi, do you consider this regression based method https://github.com/sail-sg/mvp? Thanks!

**Comment by woo1** [comment]
Great work!
Hope to support [TCMR](https://github.com/hongsukchoi/TCMR_RELEASE): [CVPR 2021] Beyond Static Features for Temporally Consistent 3D Human Pose and Shape from a Video
and [Hand4Whole](https://github.com/mks0601/Hand4Whole_RELEASE): [CVPRW 2022] Accurate 3D Hand Pose Estimation for Whole-Body 3D Human Mesh Estimation

**Comment by daixiangzi** [comment]
Hope to support cliff

**Comment by imabackstabber** [comment]
Hope to support MeshTransformer and FastMETRO.
Always overfitting when I train them from scratch.

**Comment by ShreelekhaR** [comment]
And to support HUMBI or Panoptic Studio as datasets

**Comment by Taylorminer** [comment]
Hope to support [VirtualPose](https://github.com/wkom/VirtualPose). VirtualPose: Learning Generalizable 3D Human Pose Models from Virtual Data (ECCV 2022)

**Comment by YaoBeiji** [comment]
啥时候来个4d assocaition graph 多视角多人的3d 估计

**Comment by imabackstabber** [comment]
Hope to support BEHAVE dataset, which is on cvpr 2023 workshop Rhobin challenge now.

**Comment by fpq199182** [comment]
请问有人脸3d重建相关的吗

**Comment by Theremoons** [comment]
在3D人体状态检测的模型中可以输出各个关节相关父关节的欧拉旋转角度吗？如果可以，那可以直接应用到游戏的3D人物了。
